app/src/main/java/org/hyperledger/besu/services/TransactionSelectionServiceImpl.java

```diff
@@ -14,42 +14,258 @@
+import org.hyperledger.besu.evm.operation.Operation;
+import org.hyperledger.besu.evm.worldstate.WorldView;
+import org.hyperledger.besu.plugin.data.BlockBody;
+import org.hyperledger.besu.plugin.data.BlockHeader;
 import org.hyperledger.besu.plugin.data.ProcessableBlockHeader;
+import org.hyperledger.besu.plugin.data.TransactionProcessingResult;
+import org.hyperledger.besu.plugin.data.TransactionSelectionResult;
 import org.hyperledger.besu.plugin.services.TransactionSelectionService;
+import org.hyperledger.besu.plugin.services.tracer.BlockAwareOperationTracer;
 import org.hyperledger.besu.plugin.services.txselection.BlockTransactionSelectionService;
 import org.hyperledger.besu.plugin.services.txselection.PluginTransactionSelector;
 import org.hyperledger.besu.plugin.services.txselection.PluginTransactionSelectorFactory;
 import org.hyperledger.besu.plugin.services.txselection.SelectorsStateManager;
+import org.hyperledger.besu.plugin.services.txselection.TransactionEvaluationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.tuweni.bytes.Bytes;
 
 /** The Transaction Selection service implementation. */
 public class TransactionSelectionServiceImpl implements TransactionSelectionService {
 
   /** Default Constructor. */
   public TransactionSelectionServiceImpl() {}
 
-  private PluginTransactionSelectorFactory factory = PluginTransactionSelectorFactory.NO_OP_FACTORY;
+  private List<PluginTransactionSelectorFactory> factories;
 
   @Override
   public PluginTransactionSelector createPluginTransactionSelector(
       final SelectorsStateManager selectorsStateManager) {
-    return factory.create(selectorsStateManager);
+    if (factories == null) {
+      return PluginTransactionSelector.ACCEPT_ALL;
+    }
+    return new AggregatedPluginTransactionSelector(
+        factories.stream().map(factory -> factory.create(selectorsStateManager)).toList());
   }
 
   @Override
   public void selectPendingTransactions(
       final BlockTransactionSelectionService selectionService,
       final ProcessableBlockHeader pendingBlockHeader) {
-    factory.selectPendingTransactions(selectionService, pendingBlockHeader);
+    if (factories != null) {
+      factories.forEach(
+          factory -> factory.selectPendingTransactions(selectionService, pendingBlockHeader));
+    }
   }
 
   @Override
   public void registerPluginTransactionSelectorFactory(
       final PluginTransactionSelectorFactory pluginTransactionSelectorFactory) {
-    checkState(
-        factory == PluginTransactionSelectorFactory.NO_OP_FACTORY,
-        "PluginTransactionSelectorFactory was already registered");
-    factory = pluginTransactionSelectorFactory;
+    if (factories == null) {
+      factories = new ArrayList<>();
+    }
+    factories.add(pluginTransactionSelectorFactory);
+  }
+
+  private static class AggregatedPluginTransactionSelector implements PluginTransactionSelector {
```

Comment:
intellij says this class can be a record

Reply:
I saw that, but in this case I do not see the need to make it a record